geopy/geocoders/base.py

```diff
@@ -185,6 +187,10 @@ def __init__(
                             else options.default_ssl_context)
 
         if self.proxies:
+            if isinstance(self.proxies, str):
```

Comment:
For compatibility with py2, `geopy.compat.string_compare` should be used instead of `str`.